### PR TITLE
Remove password hint responses from API

### PR DESCRIPTION
**Pull request by MGibson1**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-10906

## 📔 Objective

Password hints are intended to be readable only via emails sent by the server. Including in profile data breaks this access limitation. This PR removes the hint from any api responses that included it. It also removes deprecated means of updating the hint.

The set password component is no longer populated with the previous hint, but changing the password already implied changing the hint. This behavior is unchanged, it's just that the default new hint is blank rather than equal to the old one.

Relates to bitwarden/server#4635

## Screenshots

**Previously, hints were displayed in password change dialog**
image

**Now, it's always blank**
image

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: apps/desktop/src/main/window.main.ts
Comment (JaredSnider-Bitwarden) [inline apps/desktop/src/main/window.main.ts:1]: ❓ : Should these new logs be introduced as a separate PR since they seem unrelated?

Comment (JaredSnider-Bitwarden) [reply apps/desktop/src/main/window.main.ts:1]: Per discussion, will keep in for debugging.

**Review by rr-bw — COMMENTED** [review]
If I understood the slack thread correctly, leaving the "New master password hint" field blank will overwrite any existing hint to be undefined? If that's true, my concern is that a user will leave the field blank assuming they are only changing their master password, and not the hint, since the button does just say "Change master password". Is there a way we could check if the hint field is `""`/`undefined`, and if so, do not update the hint? If not, maybe a short descriptive text saying "Leaving the hint blank _will_ overwrite, etc..."

**Comment by MGibson1** [comment]
@rr-bw I don't think that is actually a big deal. Updating a password _should_ imply change of the hint -- why would a hint be valid for multiple?

The new label also helps clarify this update

**Review by rr-bw — APPROVED** [review]

**Review by JaredSnider-Bitwarden — APPROVED** [review]
